be/src/exprs/condition_expr.cpp

```diff
@@ -111,6 +115,26 @@ class VectorizedIfNullExpr : public Expr {
 
         return result.build(ColumnHelper::is_all_const(columns));
     }
+
+    ColumnPtr _evaluate_complex(const Columns& columns) {
+        auto num_rows = columns[0]->size();
+        auto res = ColumnHelper::cast_to_nullable_column(columns[0]->clone_empty());
+        res->reserve(num_rows);
+        NullColumnPtr null = nullptr;
+
+        if (columns[0]->is_nullable()) {
+            null = down_cast<NullableColumn*>(columns[0].get())->null_column();
+        }
+
+        for (int row = 0; row < num_rows; ++row) {
+            if (null == nullptr || !null->get_data()[row]) { // not null
+                res->append(*columns[0], row, 1);
+            } else {
+                res->append(*columns[1], row, 1);
+            }
+        }
```

Comment:
in the case of `null == nullptr`, can we use a separate loop to append columns[0] directly without having to check inside each loop?

Reply:
branch prediction will optimize that case where a branch are almost true.